docs/Training-on-Amazon-SageMaker-RL.md

```diff
@@ -0,0 +1,283 @@
+For the Basic environment, we deploy the model and pass an input to the predictor:
+
+```
+from sagemaker.tensorflow.model import TensorFlowModel
+
+model = TensorFlowModel(model_data=estimator.model_data,
+              framework_version='2.1.0',
+              role=role)
+
+predictor = model.deploy(initial_instance_count=1,
+                         instance_type=instance_type)
+input = {"inputs": {'observations': np.ones(shape=(1, 20)).tolist(),
+                    'prev_action': [0, 0],
+                    'is_training': False,
+                    'prev_reward': -1,
+                    'seq_lens': -1
+                   }
+        }
+result = predictor.predict(input)print(result['outputs']['actions'])
+```
+
+The model predicts an indicator corresponding to moving left or right. The recommended direction of movement for the blue box agent always points towards the larger green ball.
+
+## Cleaning up
+
+When you’re finished running the model, call `predictor.delete_endpoint()` to delete the model deployment endpoint to avoid incurring future charges.
+
+## (Optional) Customizing training algorithms, models, and environments
+
+In addition to the preceding use case, we encourage you to explore the customization capabilities this solution supports.
+In the preceding code example, we specify [Proximal Policy Optimization](https://openai.com/blog/openai-baselines-ppo/) (PPO) to be the training algorithm. PPO is a popular RL algorithm that performs comparably to state-of-the-art approaches but is much simpler to implement and tune. Depending on your use case, you can choose the most-fitted algorithm for training by either selecting from a list of comprehensive algorithms already implemented in RLLib or building a custom algorithm from scratch.
+By default, RLLib applies a pre-defined [convolutional neural network](https://github.com/ray-project/ray/blob/master/rllib/models/tf/visionnet.py) or [fully connected neural network](https://github.com/ray-project/ray/blob/master/rllib/models/tf/fcnet.py). However, you can create a custom model for training and testing. Following the [examples](https://github.com/ray-project/ray/blob/master/rllib/examples/custom_env.py) from RLLib, you can register the custom model by calling `ModelCatalog.register_custom_model`, then refer to the newly registered model using the `custom_model` argument.
+In our code example, we invoke a predefined Unity environment called Basic, but you can experiment with other [pre-built Unity environments](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Learning-Environment-Examples.md). However, as of this writing, our solution only supports a single-agent environment. When new environments are built, register it by calling `register_env` and refer to the environment with the env parameter.
```

Comment:
I would put this note around single-agent environment as a separate disclaimer or move it up in the document.